java/statestore/src/main/java/sleeper/statestore/dynamodb/DynamoDBFileReferenceStore.java

```diff
@@ -127,6 +128,86 @@ public void addFiles(List<FileReference> fileReferences) throws StateStoreExcept
         }
     }
 
+    @Override
+    public void splitFileReferences(List<SplitFileReferenceRequest> splitRequests) throws StateStoreException {
+        long updateTime = clock.millis();
+        List<SplitFileReferenceRequest> batchRequests = new ArrayList<>();
+        List<TransactWriteItem> batchReferenceWrites = new ArrayList<>();
+        Map<String, Integer> batchReferenceCountUpdates = new HashMap<>();
+        int firstUnappliedRequestIndex = 0;
+        try {
+            for (int i = 0; i < splitRequests.size(); i++) {
+                SplitFileReferenceRequest splitRequest = splitRequests.get(i);
+                List<TransactWriteItem> requestWrites = splitFileReferenceWrites(splitRequest, updateTime);
+                int newBatchWrites = batchReferenceWrites.size() + requestWrites.size() + batchReferenceCountUpdates.size();
+                if (!batchReferenceCountUpdates.containsKey(splitRequest.getFilename())) {
+                    // Reference count updates need to be aggregated into one for each file, so this will only result in
+                    // a separate write item if this file has not been updated by a previous request
+                    newBatchWrites += 1;
+                }
+                if (newBatchWrites > 100) {
+                    applySplitRequestWrites(batchRequests, batchReferenceWrites, batchReferenceCountUpdates, updateTime);
+                    firstUnappliedRequestIndex = i;
+                    batchReferenceWrites.clear();
+                    batchRequests.clear();
+                }
+                batchReferenceWrites.addAll(requestWrites);
+                batchRequests.add(splitRequest);
+                int referenceCountDiff = splitRequest.getNewReferences().size() - 1;
+                batchReferenceCountUpdates.compute(splitRequest.getFilename(), (file, update) ->
+                        update == null ? referenceCountDiff : update + referenceCountDiff);
```

Comment:
Add logging to a few places in this loop so that the user can understand what's going on.

Reply:
I've added a couple of logs to differentiate the cases where split reference requests are applied.